cpg_infra/driver.py

```diff
@@ -240,68 +461,196 @@ def deployment_accounts_by_access_level(self):
         return {k: v for k, v in accounts.items() if v}
 
     # endregion MACHINE ACCOUNTS
+
+    # region PERSON ACCESS
+    def setup_externally_specified_members(self):
+
+        if not INFRA_MEMBERS_PATH:
+            return
+
+        if not os.path.exists(INFRA_MEMBERS_PATH):
+            raise ValueError(
+                f'Could not find the "cpg-infrastructure-private" repo at: '
+                f'{INFRA_MEMBERS_PATH}, consider setting the '
+                f'"INFRA_MEMBERS_PATH" environment variable'
+            )
+
+        groups = [
+            self.data_manager_group,
+            self.analysis_group,
+            self.metadata_access_group,
+            self.upload_group,
+            self.web_access_group,
+        ]
+
+        filepath = os.path.join(
+            INFRA_MEMBERS_PATH, f'{self.dataset_config.dataset}/members.yaml'
+        )
+
+        if not os.path.exists(filepath):
+            return
+
+        with open(filepath, encoding='utf-8') as f:
+            d = yaml.safe_load(f)
+
+        def compute_hash(s):
+            # add a salt so you can't track members through the resource names
+            # between different stacks - a small obfuscation.
+            return xxhash.xxh32((self.dataset_config.dataset + s).encode()).hexdigest()
+
+        for group in groups:
+            group_name = group.name.removeprefix(self.dataset_config.dataset + '-')
+            for member in d.get(group_name, []):
+                h = compute_hash(member)
```

Comment:
Do we want to use a more readable (but nevertheless sanitized) version here? Something like mapping `micfra@mail.com.au` --> `m****a@mail.com.au` or similar? The deployment diffs would become more useful that way.

Reply:
I've switched this to use the initials, followed by the hash to avoid clashes and promote readability